blog/services/pin_service.py

```
@@ -0,0 +1,98 @@
+class NotPinnedError(PinError):
+    pass
+
+
+class InvalidPinOrderError(PinError):
+    pass
+
+
+@transaction.atomic
+def add_pin(post, max_count: int):
+    if max_count < 1:
+        raise ValueError("max_count must be greater than 0")
```

Ruff (0.15.6)

[warning] Avoid specifying long messages outside the exception class

(TRY003)